src/modules/featureform/featureform-directive.js

```diff
@@ -61,76 +61,78 @@ angular.module('anol.featureform')
+
+                    scope.formFields = [];
+                    scope.properties = {};
+
+                    var propertyWatchers = [];
+
+                    /**
+                     * @param {ol.Feature} feature
+                     */
+                    var featureChangeHandler = function (feature) {
+                        clearPropertiesWatchers();
+                        if (feature) {
+                            scope.properties = feature.getProperties();
+                            const geomType = feature.getGeometry().getType();
+                            if (geomType === 'Point') {
+                                scope.formFields = scope.pointFields;
+                            } else if (geomType === 'LineString') {
+                                scope.formFields = scope.lineFields;
+                            } else if (geomType === 'Polygon') {
+                                scope.formFields = scope.polygonFields;
+                            }
+                            setupPropertiesWatchers(scope.formFields);
+                        }
+                    };
+
+                    /**
+                     * @param {FormField[]} formFields
+                     */
+                    var setupPropertiesWatchers = function (formFields) {
+                        formFields.forEach(function (field) {
+                            var watcher = scope.$watch('properties["' + field.name + '"]', function (value) {
+                                scope.feature.set(field.name, value);
+                            });
+                            propertyWatchers.push(watcher);
+                        });
+                    };
+
+                    var clearPropertiesWatchers = function () {
+                        propertyWatchers.forEach(function (dewatch) {
+                            dewatch();
+                        });
+                        propertyWatchers = [];
+                    };
+
+                    /**
+                     * @param {SelectOption|string} option
+                     * @return {string}
+                     */
+                    scope.getOptionValue = function (option) {
+                        return angular.isObject(option) ? option.value : option;
+                    };
+
+                    /**
+                     * @param {SelectOption|string} option
+                     * @return {string}
+                     */
+                    scope.getOptionLabel = function (option) {
+                        return angular.isObject(option) ? option.label || option.value : option;
+                    };
+
+                    /**
+                     * @param {FormField} field
+                     * @return {boolean}
+                     */
+                    scope.isValid = function (field) {
+                        if (field.required) {
+                            return !!scope.properties[field.name];
```

Comment:
Does this make a form invalid, if a number field contains a `0`?

Reply:
good point